src/puppetlabs/trapperkeeper/core.clj

```diff
@@ -173,10 +173,12 @@
           (internal/parse-cli-args!)
           (run))
       (catch map? m
-        (case (without-ns (:type m))
-          :cli-error (quit 1 (:message m) *err*)
-          :cli-help (quit 0 (:message m) *out*)
-          (throw+)))
+        (let [type (:type m)]
+          (if (keyword? type)
```

Comment:
I think there are some others that could be caught here, yeah.  I guess the question for those would be whether it would be preferable for the user to see a simple message as opposed to a full stack trace with the error.  I think the `cli-error` and `cli-help` ones are caught specifically because it is known that the short message they contain is probably all the user would need in order to understand the text whereas with anything else we don't necessarily know what the user might think is interesting to see so we just let the whole stack trace be dumped out.  If we were to put an `:else` here, what do you think it should have in it?